go/libkb/wot.go

```diff
@@ -198,48 +202,66 @@ func fetchWot(mctx MetaContext, username *string) (res []serverWotVouch, err err
+	vouches, err := fetchWot(mctx, &vouchee, voucher)
 	if err != nil {
-		mctx.Debug("error fetching pending web-of-trust vouches: %s", err.Error())
+		mctx.Debug("error fetching web-of-trust vouches for %s: %s", vouchee, err.Error())
 		return nil, err
 	}
-	me, err := LoadMe(NewLoadUserArgWithMetaContext(mctx))
-	if err != nil {
-		return nil, fmt.Errorf("error loading myself: %s", err.Error())
+	if voucheeUser == nil {
+		voucheeUser, err = LoadUser(NewLoadUserArgWithMetaContext(mctx).WithName(vouchee))
+		if err != nil {
+			return nil, fmt.Errorf("error loading vouchee: %s", err.Error())
+		}
 	}
-	for _, serverVouch := range serverVouches {
-		vouch, err := transformUserVouch(mctx, serverVouch, me)
+	for _, serverVouch := range vouches {
+		vouch, err := transformUserVouch(mctx, serverVouch, voucheeUser)
 		if err != nil {
-			mctx.Debug("error validating server-reported pending web-of-trust vouches: %s", err.Error())
+			mctx.Debug("error validating server-reported web-of-trust vouches for %s: %s", vouchee, err.Error())
 			return nil, err
 		}
 		res = append(res, vouch)
 	}
-	mctx.Debug("found %d web-of-trust vouches", len(res))
+	mctx.Debug("found %d web-of-trust vouches for %s", vouchee, len(res))
 	return res, nil
 }
 
-func FetchUserWot(mctx MetaContext, username string) (res []keybase1.WotVouch, err error) {
-	defer mctx.Trace("FetchUserWot", func() error { return err })()
-	vouches, err := fetchWot(mctx, &username)
+// FetchMyWot gets all vouches (of any status) written for me
+func FetchMyWot(mctx MetaContext) (res []keybase1.WotVouch, err error) {
+	defer mctx.Trace("FetchMyWot", func() error { return err })()
+	me, err := LoadMe(NewLoadUserArgWithMetaContext(mctx))
 	if err != nil {
-		mctx.Debug("error fetching web-of-trust vouches for %s: %s", username, err.Error())
-		return nil, err
+		return nil, fmt.Errorf("error loading myself: %s", err.Error())
 	}
-	vouchee, err := LoadUser(NewLoadUserArgWithMetaContext(mctx).WithName(username))
+	myName := me.GetName()
+	return fetchUserWotByVoucher(mctx, myName, nil, me)
+}
+
+// FetchMyWotByUser gets vouch (of any status) written for me by the voucher
+func FetchMyWotByUser(mctx MetaContext, voucher string) (res []keybase1.WotVouch, err error) {
+	defer mctx.Trace("FetchMyWotByUser", func() error { return err })()
+	me, err := LoadMe(NewLoadUserArgWithMetaContext(mctx))
 	if err != nil {
-		return res, fmt.Errorf("error loading vouchee: %s", err.Error())
-	}
-	for _, serverVouch := range vouches {
-		vouch, err := transformUserVouch(mctx, serverVouch, vouchee)
-		if err != nil {
-			mctx.Debug("error validating server-reported web-of-trust vouches for %s: %s", username, err.Error())
-			return nil, err
-		}
-		res = append(res, vouch)
+		return nil, fmt.Errorf("error loading myself: %s", err.Error())
 	}
-	mctx.Debug("found %d web-of-trust vouches for %s", username, len(res))
-	return res, nil
+	myName := me.GetName()
+	return fetchUserWotByVoucher(mctx, myName, &voucher, me)
```

Comment:
need to explicitly pass in vouchee (which is me), because if vouchee==nil and voucher==me, then the server will return all vouches written by voucher/me, which is not what `FetchMyWotByUser` expects.